src/model/index.js

```diff
@@ -88,7 +87,9 @@ export default function(fields) {
       [fields, params] = [RETURNS_ALL_FIELDS, fields];
     }
     return new Query(model, fields, RETURNS_LIST, params);
-  }
+  };
+
+  model.modelName = name;
```

Comment:
`read-only` is a really astute point — we shouldn't ever be able to change this after it's defined. We should probably use a `getter` here in this case: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get.

We'll probably need to use `defineProperty` to create it though :(